File: Compiler/src/main/kotlin/nodes/VarItemNode.java
package nodes;

public class VarItemNode {
    public VarType type = VarType.UNINITIALIZED; // Тип переменной

    public int id = 0;

    public boolean is_map_key = false; // Получена ли переменная по ключу (через точку)

    public String ident = ""; // Переменная либо имя ключа

    public ExprNode first_expr = null; // Function call или expr в скобках
    public ExprNode second_expr = null; // expr в квадратных скобках

    public VarItemNode next = null;
}

File: Compiler/src/main/kotlin/nodes/VarItemNode.kt
package nodes

class VarItemNode {
    var type = VarType.UNINITIALIZED // Тип переменной
    var id = 0
    var isMapKey = false // Получена ли переменная по ключу (через точку)
    var ident = "" // Переменная либо имя ключа
    var firstExpr: ExprNode? = null // Function call или expr в скобках
    var secondExpr: ExprNode? = null // expr в квадратных скобках
    var next: VarItemNode? = null
}